arangod/IResearch/IResearchKludge.cpp

```diff
@@ -39,6 +42,28 @@ inline void normalizeExpansion(std::string& name) {
 
 }  // namespace
 
+namespace arangodb {
+void syncIndexOnCreate(Index* index) {
+  TRI_ASSERT(index);
+  iresearch::IResearchDataStore* store{nullptr};
+  switch (index->type()) {
+    case Index::IndexType::TRI_IDX_TYPE_IRESEARCH_LINK:
+      store = basics::downCast<iresearch::IResearchDataStore>(
+          basics::downCast<iresearch::IResearchRocksDBLink>(index));
+      break;
+    case Index::IndexType::TRI_IDX_TYPE_INVERTED_INDEX:
+      store = basics::downCast<iresearch::IResearchDataStore>(
```

Comment:
Remove downCast to DataStore, that is upcast, it can be implicit, and i think here its ok